speculos/client.py

```diff
@@ -185,15 +185,19 @@ def stop(self) -> None:
 
         self.process = None
 
-    def __del__(self) -> None:
-        self.stop()
-
 
 class SpeculosClient(Api, SpeculosInstance):
     def __init__(self, app: str, args: List[str] = [], api_url: str = "http://127.0.0.1:5000") -> None:
         SpeculosInstance.__init__(self, app, args)
+        self.__api_url = api_url
+
+    def __enter__(self) -> "SpeculosClient":
         SpeculosInstance.start(self)
-        Api.__init__(self, api_url)
+        Api.__init__(self, self.__api_url)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:
```

Comment:
Please add type annotation for all arguments.

FYI, it was found with:

```console
$ mypy --strict speculos/client.py
speculos/client.py:199: error: Function is missing a type annotation for one or more arguments
```

unfortunately `mypy` warnings should be fixed across the entire code base before being integrated to the CI.